pkg/providers/imagefamily/nodeimage_test.go

```diff
@@ -174,63 +184,155 @@ var _ = Describe("NodeImageProvider tests", func() {
-			nodeClass.Spec.ImageFamily = lo.ToPtr(v1beta1.AzureLinuxImageFamily)
-			nodeClass.Status.KubernetesVersion = "1.32.0"
-
-			foundImages, err := nodeImageProvider.List(ctx, nodeClass)
-			Expect(err).ToNot(HaveOccurred())
-			Expect(foundImages).To(ContainElements(getExpectedTestSIGImages(*nodeClass.Spec.ImageFamily, sigImageVersion, "1.32.0")))
-
-			// Explicitly verify ARM64 image IS included in SIG (Shared Image Gallery)
-			Expect(foundImages).To(ContainElement(HaveField("ID", ContainSubstring("V3gen2arm64"))))
+		Context("List Default Images When FIPSMode Is Explicitly Disabled", func() {
+			BeforeEach(func() {
+				nodeClass.Spec.FIPSMode = &v1beta1.FIPSModeDisabled
+			})
+
+			It("should match expected images for generic Ubuntu", func() {
+				nodeClass.Spec.ImageFamily = lo.ToPtr(v1beta1.UbuntuImageFamily)
+				foundImages, err := nodeImageProvider.List(ctx, nodeClass)
+				Expect(err).ToNot(HaveOccurred())
+				expectedImages := getExpectedTestSIGImages(*nodeClass.Spec.ImageFamily, nodeClass.Spec.FIPSMode, sigImageVersion, kubernetesVersion)
+				Expect(foundImages).To(Equal(expectedImages))
+
+			})
+
+			It("should match expected images for Ubuntu2204", func() {
+				nodeClass.Spec.ImageFamily = lo.ToPtr(v1beta1.Ubuntu2204ImageFamily)
+				foundImages, err := nodeImageProvider.List(ctx, nodeClass)
+				Expect(err).ToNot(HaveOccurred())
+				expectedImages := getExpectedTestSIGImages(*nodeClass.Spec.ImageFamily, nodeClass.Spec.FIPSMode, sigImageVersion, kubernetesVersion)
+				Expect(foundImages).To(Equal(expectedImages))
+			})
+
+			// This test changes depending on the Kubernetes version, in effect making version-specific tests unnecessary.
+			// They are still kept for clarity and to ensure that the behavior is explicitly tested.
+			It("should match expected images for default AzureLinux, depending on the Kubernetes version", func() {
+				nodeClass.Spec.ImageFamily = lo.ToPtr(v1beta1.AzureLinuxImageFamily)
+				foundImages, err := nodeImageProvider.List(ctx, nodeClass)
+				Expect(err).ToNot(HaveOccurred())
+				expectedImages := getExpectedTestSIGImages(*nodeClass.Spec.ImageFamily, nodeClass.Spec.FIPSMode, sigImageVersion, kubernetesVersion)
+				Expect(foundImages).To(Equal(expectedImages))
+			})
 		})
+
+		DescribeTable("should match expected images",
+			func(imageFamily *string, fipsMode *v1beta1.FIPSMode, version string, kubernetesVersion string) {
+				nodeClass.Spec.ImageFamily = imageFamily
+				nodeClass.Spec.FIPSMode = fipsMode
+				nodeClass.Status.KubernetesVersion = kubernetesVersion
+
+				foundImages, err := nodeImageProvider.List(ctx, nodeClass)
+				Expect(err).ToNot(HaveOccurred())
+				expectedImages := getExpectedTestSIGImages(*nodeClass.Spec.ImageFamily, nodeClass.Spec.FIPSMode, sigImageVersion, kubernetesVersion)
+				Expect(foundImages).To(Equal(expectedImages))
+        
+        // TODO: Improve direct string comparison
+        if kubernetesVersion == "1.32.0" {
```

Comment:
Can handle this with `imagefamily.UseAzureLinux3`:
```suggestion
        if imagefamily.UseAzureLinux3(kubernetesVersion) {
```